### PR TITLE
add build & runtime debug logging

### PR DESCRIPTION
**Pull request by axe312ger**
### Description

Adds support for the debug module to the project.

Via `DEBUG=docz*` you now get:

* While building:
  * The actual file matching patterns applied
  * The matched files from these patterns
  * The list of filtered components
* In Browser:
  * A list of components available in state
  * The pattern matched for locating the correct component

It simplifies the component matching in the Props component as well.

Post merge todos:

* https://github.com/doczjs/docz-website/issues/91

### CODE REVIEW

**Comment by axe312ger** [comment]
@rakannimer I struggle to run the e2e tests in dev mode on my machine. Do i need to start the basic example as server or sth?

**Review by rakannimer — DISMISSED** [review]

**Comment by rakannimer** [comment]
It looks like the props are not being rendered properly in the basic example, it seems something broke in the docz package here : https://github.com/doczjs/docz/pull/1321/files#diff-5cc2196461930cb24939fb44764d656bL121 


How are e2e-tests failing on your setup ? Did you try running 

```sh
yarn packages && node other-packages/index.js
```
You can also try running the basic example with the dev-env by running : 

```sh
yarn
cd dev-env/basic
yarn
node dev.js
```

The alert props should be missing if the test output is correct.

**Comment by axe312ger** [comment]
Both fail 😅 

> ```shell
> yarn packages && node other-packages/index.js
> ```

```
...
docz-core: Test Suites: 4 passed, 4 total
docz-core: Tests:       16 passed, 16 total
docz-core: Snapshots:   0 total
docz-core: Time:        6.546s, estimated 8s
docz-core: Ran all test suites.
docz-core: ⚠  warning   No cache was found with your props definitions
docz-core: ⚠  warning   We'll parse your components to get props from them
docz-core: ⚠  warning   Depending on your components, this could take while...
lerna success run Ran npm script 'test' in 4 packages in 8.1s:
lerna success - docz-core
lerna success - rehype-docz
lerna success - remark-docz
lerna success - babel-plugin-export-metadata
✨  Done in 43.03s.
internal/modules/cjs/loader.js:716
    throw err;
    ^

Error: Cannot find module '/Users/bene/dev/opensource/docz/other-packages/index.js'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:713:15)
    at Function.Module._load (internal/modules/cjs/loader.js:618:27)
    at Function.Module.runMain (internal/modules/cjs/loader.js:931:10)
    at internal/main/run_main_module.js:17:11 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
```

> You can also try running the basic example with the dev-env by running :
> 
> ```shell
> yarn
> cd dev-env/basic
> yarn
> node dev.js
> ```


```
[2019-12-16 11:20:52] waiting for changes...

Building app

 ERROR #10123  CONFIG

We encountered an error while trying to load your site's gatsby-config. Please fix the error and try again.



  TypeError: mergeWith is not a function
```

**Comment by axe312ger** [comment]
when executing `e2e-tests`:

```
$ node index.js
Running npx verdaccio
Waiting for Verdaccio to boot
npm ERR! code ECONNREFUSED
npm ERR! errno ECONNREFUSED
npm ERR! FetchError: request to http://localhost:4873/verdaccio failed, reason: connect ECONNREFUSED 127.0.0.1:4873
npm ERR!     at ClientRequest.<anonymous> (/Users/bene/.nvm/versions/node/v12.8.1/lib/node_modules/npm/node_modules/node-fetch-npm/src/index.js:68:14)
npm ERR!     at ClientRequest.emit (events.js:203:13)
npm ERR!     at Socket.socketErrorListener (_http_client.js:399:9)
npm ERR!     at Socket.emit (events.js:203:13)
npm ERR!     at emitErrorNT (internal/streams/destroy.js:91:8)
npm ERR!     at emitErrorAndCloseNT (internal/streams/destroy.js:59:3)
npm ERR!     at processTicksAndRejections (internal/process/task_queues.js:77:11)
npm ERR!  FetchError: request to http://localhost:4873/verdaccio failed, reason: connect ECONNREFUSED 127.0.0.1:4873
npm ERR!     at ClientRequest.<anonymous> (/Users/bene/.nvm/versions/node/v12.8.1/lib/node_modules/npm/node_modules/node-fetch-npm/src/index.js:68:14)
npm ERR!     at ClientRequest.emit (events.js:203:13)
npm ERR!     at Socket.socketErrorListener (_http_client.js:399:9)
npm ERR!     at Socket.emit (events.js:203:13)
npm ERR!     at emitErrorNT (internal/streams/destroy.js:91:8)
npm ERR!     at emitErrorAndCloseNT (internal/streams/destroy.js:59:3)
npm ERR!     at processTicksAndRejections (internal/process/task_queues.js:77:11) {
npm ERR!   message: 'request to http://localhost:4873/verdaccio failed, reason: connect ECONNREFUSED 127.0.0.1:4873',
npm ERR!   type: 'system',
npm ERR!   errno: 'ECONNREFUSED',
npm ERR!   code: 'ECONNREFUSED',
npm ERR!   stack: 'FetchError: request to http://localhost:4873/verdaccio failed, reason: connect ECONNREFUSED 127.0.0.1:4873\n' +
npm ERR!     '    at ClientRequest.<anonymous> (/Users/bene/.nvm/versions/node/v12.8.1/lib/node_modules/npm/node_modules/node-fetch-npm/src/index.js:68:14)\n' +
npm ERR!     '    at ClientRequest.emit (events.js:203:13)\n' +
npm ERR!     '    at Socket.socketErrorListener (_http_client.js:399:9)\n' +
npm ERR!     '    at Socket.emit (events.js:203:13)\n' +
npm ERR!     '    at emitErrorNT (internal/streams/destroy.js:91:8)\n' +
npm ERR!     '    at emitErrorAndCloseNT (internal/streams/destroy.js:59:3)\n' +
npm ERR!     '    at processTicksAndRejections (internal/process/task_queues.js:77:11)'
npm ERR! }
npm ERR!
npm ERR! If you are behind a proxy, please make sure that the
npm ERR! 'proxy' config is set properly.  See: 'npm help config'
```

**Comment by rakannimer** [comment]
Could you pull from master and try running the dev-env again ? There was an outdated dependency that should be fixed now.

Not sure what's happening when you're running e2e-tests 🤔

**Comment by axe312ger** [comment]
Nice thanks! The basic demo is running now! Can reproduce the issue. Will check and push a fix :)

File: core/docz/src/components/Props.tsx [hunk 1]
```diff
@@ -118,20 +121,28 @@ export const Props: SFC<PropsProps> = ({
...
+    `/${componentName}.`,
+    `/${kebabCase(componentName)}.`,
+    `/${pascalCase(componentName)}.`,
+  ]
+
+  debug(`Matching component props via ${matchingPatterns.join(', ')}:`)
+  debug(keys)
+
+  const found = keys.find(key => {
+    return !!matchingPatterns.find(pattern => key.includes(pattern))
```
Comment (lucalves) [inline core/docz/src/components/Props.tsx]: CircleCI is probably breaking here. I would recommend checking the PR again and analyzing where exactly it is breaking.

**Comment by axe312ger** [comment]
Hey there.

I moved on and I am no more using docz in my projects. Therefore I have to time to work on this PR anymore.

Feel free to take it over, otherwise just close it

**Pull request closed without merging**